### PR TITLE
boards: arm: disco_l475_iot1: add stm32cubeprogrammer

### PR DESCRIPTION
**Pull request by Desvauxm-st**
add stm32cubeprogrammer runner for disco_l475_iot1 boards

Since https://github.com/zephyrproject-rtos/zephyr/pull/62647, debug in stop mode is explicitely disabled in f/w when debug is not enabled. 
After running application with PM enabled, it is not possible to flash using openocd. One consequence is tests failed in CI when openocd is used. However we can still use flash using stm32cubeprogrammer runner, so add stm32cubeprogrammer to this board in order to use it with twister.

### CODE REVIEW

**Review by erwango — APPROVED** [review]

File: boards/arm/disco_l475_iot1/board.cmake [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
+board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
 board_runner_args(jlink "--device=STM32L475VG" "--speed=4000")
 
+include(${ZEPHYR_BASE}/boards/common/stm32cubeprogrammer.board.cmake)
```
Comment (erwango) [inline boards/arm/disco_l475_iot1/board.cmake:6]: Please put this line in last position

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/disco_l475_iot1/board.cmake (hunk 1 above)
Comment (Desvauxm-st) [reply boards/arm/disco_l475_iot1/board.cmake]: ok

**Review by erwango — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
Wouldn't more stm32 board be affected as well and need a similar workaround? Also, is there any openocd `reset_config` option that would work in this case?

**Comment by erwango** [comment]
> Wouldn't more stm32 board be affected as well and need a similar workaround? Also, is there any openocd reset_config option that would work in this case?

This PR is indeed not the best option and we'll work on a wider impact change, but considered the release timeline we wanted to add this no impact change and to at least get correct bench results.

**Review by fabiobaltieri — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**